### ISSUE TITLE
AsyncAdvancedCircuitBreakerTResultSyntax not Returning AsyncPolicy{TResult}

### ISSUE THREAD

**Issue by diegodancourt** [App-vNext/Polly#586]
**Summary**: 
On version 7.02 of Polly AdvancedCircuitBreakerAsync(..) doesn't return an AsyncCircuitBreakerPolicy like it's non-advanced counterpart CircuitBreakerAsync.

<br/>
<br/>

**Expected behavior**: 

The following should compile:

**EDITED** by @reisenberger to show correct syntax:

```
IAsyncPolicy<HttpResponseMessage> myHttpApplyCircuitBreakerPolicy =
                Policy.HandleResult<HttpResponseMessage>(r => !r.IsSuccessStatusCode).AdvancedCircuitBreakerAsync(...)
```

<br/>
<br/>

**Actual behavior**:
The above code doesn't compile because of the mixed returned types. It used to compile in version 6.1.2.

<br/>
<br/>

**Comment by reisenberger** [comment]
Thanks @diegodancourt .  This will be fixed in #589

**Comment by reisenberger** [comment]
Closed by release of v7.0.3
